src/elements/button-link/src/index.tsx

```diff
@@ -1,36 +1,41 @@
 /** @jsx jsx */
 
 import * as React from 'react'
-import { jsx } from 'theme-ui'
+import { jsx, Styled } from 'theme-ui'
 
-interface Props extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
-  variant: string
-  href?: string
-}
+type Overwrite<T, U> = Omit<T, keyof U> & U
+type Props<T extends React.ComponentType<any>> = Overwrite<
+  React.ComponentProps<T>,
+  {
+    as?: T
+    variant: string
+    children: React.ReactNode
+  }
+>
 
-export const ButtonLink: React.FC<Props> = ({
+export const ButtonLink = <
+  T extends React.ComponentType<any> = React.ComponentType<
+    JSX.IntrinsicElements['a']
+  >
+>({
   children,
   variant,
-  href,
   ...props
-}) => {
-  return (
-    <a
-      sx={{
-        cursor: 'pointer',
-        backgroundImage: 'none',
-        fontFamily: 'base',
-        fontSize: 'base',
-        paddingX: 'sm',
-        paddingY: 'base',
-        display: 'inline-block',
-        textDecoration: 'none',
-        variant: `buttons.variants.${variant}`
-      }}
-      href={href}
-      {...props}
-    >
-      {children}
-    </a>
-  )
-}
+}: Props<T>) => (
+  <Styled.a
```

Comment:
In my opinion, if this looks like a button, it should be `Styled.button`, not `Styled.a` - adding a `button` property to the `styles` object in the theme should make it available. Some of the styles below could be moved to it, as well

Reply:
I thought that, but theme-ui's `Styled` doesn't have a button.

```
src/elements/button-link/src/index.tsx:25:11 - error TS2339: Property 'button' does not exist on type '{ p: SxComponent<ClassAttributes<HTMLParagraphElement> & HTMLAttributes<HTMLParagraphElement> & SxProps>; ... 27 more ...; root: SxComponent<...>; } & SxComponent<...>'.
```

Is it supposed to have it?

Reply:
Oh, that's weird! It's specifically mentioned in the docs as not available, but it doesn't explain why: https://theme-ui.com/theme-spec#styles

Carry on, then 😅